tools/cpp/cpp20modules_tools/BUILD

```diff
@@ -0,0 +1,96 @@
+# Copyright 2018 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+filegroup(
+    name = "srcs",
+    srcs = glob(
+        ["**"],
+    ),
+    visibility = ["//tools/cpp:__pkg__"],
+)
+
+cc_library(
+    name = "common",
+    srcs = ["common/common.cc"],
+    hdrs = ["common/common.h"],
+    includes = ["."],
+    deps = [
+        "@nlohmann_json//:json",
```

Comment:
This is a dependency of @bazel_tools and not of the @bazel repository, so `bazel_dep` should go into https://github.com/bazelbuild/bazel/blob/master/src/MODULE.tools

But please consider avoiding adding a dependency (even in the case this means sligltly less readable code).

We should have a JSON parses somewhere already, because in Starlark there's a `json` module.

Reply:
The JSON parser in Java is here: /src/main/java/net/starlark/java/lib/json/Json.java
In this case it seems we decided it's better to implement the parser from scratch, instead of adding a dependency. Feel free to do the same in C++.